pyiceberg/table/__init__.py

```diff
@@ -2751,10 +2824,12 @@ def _parquet_files_to_data_files(table_metadata: TableMetadata, file_paths: List
 
 class _MergingSnapshotProducer(UpdateTableMetadata["_MergingSnapshotProducer"]):
```

Comment:
I feel that `_SnapshotProducer` may be more accurate here since it does not merge manifests. WDYT?

Reply:
+1 sounds correct to rename it